acir/src/circuit/opcodes.rs

```diff
@@ -4,27 +4,91 @@ use super::directives::{Directive, LogInfo};
 use crate::native_types::{Expression, Witness};
 use crate::serialization::{read_n, read_u16, read_u32, write_bytes, write_u16, write_u32};
 use crate::BlackBoxFunc;
+use acir_field::FieldElement;
 use serde::{Deserialize, Serialize};
 
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Hash, Copy, Default)]
 pub struct BlockId(pub u32);
 
 /// Operation on a block
 /// We can either write or read at a block index
-#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
 pub struct MemOp {
     pub operation: Expression,
     pub index: Expression,
     pub value: Expression,
 }
 
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MemoryBlock {
+    pub id: BlockId,
+    pub len: u32,          //length of the memory block
```

Comment:
Note: for comments, lets put them ontop of the field with three slashes.

From:

```rust
pub len : u32, // length of the memory block 
```

to:

```rust
/// Length of the memory block
pub len : u32 
```

The latter one also shows up in your IDE when you hover over the variable in another file